src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/QualifiedNameSyntaxExtensions.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.CodeAnalysis.CSharp.Extensions
+{
+    internal static class QualifiedNameSyntaxExtensions
+    {
+        public static NameSyntax GetLeftmostName(this QualifiedNameSyntax qualifiedName)
+        {
+            var result = qualifiedName.Left;
+
+            while (result is QualifiedNameSyntax q)
```

Comment:
unclear why this only walks down qualified name.  and not, for example, alias names.  needs comment to that effect.

if it's not actually getting the true left-most name, i would not make this an extension and would move it to where the caller uses it.